src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,86 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (isset($options['extra']['curl']) && \is_array($options['extra']['curl'])) {
+            $curloptsToConfig = [
+                //options used in CurlHttpClient
+                \CURLOPT_HTTPAUTH => 'auth_ntlm',
+                \CURLOPT_USERPWD => 'auth_ntlm',
+                \CURLOPT_RESOLVE => 'resolve',
+                \CURLOPT_NOSIGNAL => 'timeout',
+                \CURLOPT_HTTPHEADER => 'headers',
+                \CURLOPT_INFILE => 'body',
+                \CURLOPT_READFUNCTION => 'body',
+                \CURLOPT_INFILESIZE => 'body',
+                \CURLOPT_POSTFIELDS => 'body',
+                \CURLOPT_UPLOAD => 'body',
+                \CURLOPT_PINNEDPUBLICKEY => 'peer_fingerprint',
+                \CURLOPT_UNIX_SOCKET_PATH => 'bindto',
+                \CURLOPT_INTERFACE => 'bindto',
+                \CURLOPT_TIMEOUT_MS => 'max_duration',
+                \CURLOPT_MAXREDIRS => 'max_redirects',
+                \CURLOPT_PROXY => 'proxy',
+                \CURLOPT_NOPROXY => 'no_proxy',
+                \CURLOPT_SSL_VERIFYPEER => 'verify_peer',
+                \CURLOPT_SSL_VERIFYHOST => 'verify_host',
+                \CURLOPT_CAINFO => 'cafile',
+                \CURLOPT_CAPATH => 'capath',
+                \CURLOPT_SSL_CIPHER_LIST => 'ciphers',
+                \CURLOPT_SSLCERT => 'local_cert',
+                \CURLOPT_SSLKEY => 'local_pk',
+                \CURLOPT_KEYPASSWD => 'passphrase',
+                \CURLOPT_CERTINFO => 'capture_peer_cert_chain',
+                \CURLOPT_USERAGENT => 'normalized_headers',
+                \CURLOPT_REFERER => 'headers',
+                //options used in CurlResponse
+                \CURLOPT_NOPROGRESS => 'on_progress',
+                \CURLOPT_PROGRESSFUNCTION => 'on_progress',
+                \CURLINFO_REDIRECT_COUNT => 'max_redirects',
+            ];
+
+            $curloptsFromResponse = [
+                \CURLOPT_PRIVATE,
+                \CURLOPT_HEADERFUNCTION,
+                \CURLOPT_WRITEFUNCTION,
+                \CURLOPT_VERBOSE,
+                \CURLOPT_STDERR,
+            ];
+
+            $methodOpts = [
+                \CURLOPT_POST,
+                \CURLOPT_PUT,
+                \CURLOPT_CUSTOMREQUEST,
+                \CURLOPT_HTTPGET,
+            ];
+
+            foreach ($options['extra']['curl'] as $opt => $optValue) {
+                if (isset($curloptsToConfig[$opt])) {
+                    $constName = $this->findConstantName($opt) ?? $opt;
+                    throw new InvalidArgumentException(sprintf('Could not override "%s" param, please use "%s" option.', $constName, $curloptsToConfig[$opt]));
+                }
+
+                if (\in_array($opt, $methodOpts)) {
+                    throw new InvalidArgumentException('Could not override method using extra.curl options.');
```

Comment:
The HTTP method cannot be overridden using "extra.curl".

Reply:
done.